### ISSUE TITLE
[Feature request] Eager load relationships 

### ISSUE THREAD

**Issue by iget-esoares** [babenkoivan/scout-elasticsearch-driver#23]
Hi,

I've read the entire docs but found no mention on how to eager-load relationships.

I've tried to use `::with` or `->with` with no success.

I think that's an important feature, if you aprove, I can implement and do a PR.

**Comment by babenkoivan** [comment]
Hi @iget-esoares,

I added the feature you have requested, now you can use it like this:

```php
App\MyModel::search('phone') 
    ->with('makers')
    ->get();
```

**Comment by iget-esoares** [comment]
Thank you! :-)

**Comment by Alexwijn** [comment]
Unfortunately this does not work when paginating. The `get` method will not be called then.

**Comment by babenkoivan** [comment]
@Alexwijn thank you for pointing that out! You are absolutely right! I'll fix that

**Comment by iget-master** [comment]
I confirm that it's impossible to eagerload when paginating :-(

**Comment by babenkoivan** [comment]
@iget-esoares hey, for some reason this issue slipped from my attention. Now it's fixed in v3.2.1.

**Comment by iget-master** [comment]
@babenkoivan Love you man! :+1: Thanks
